src/core/launcher/index.ts

```diff
@@ -133,6 +133,16 @@ export class Launcher extends Observable implements DefaultLauncher {
       }
     }
 
+    const isAddingWhoIsOnline = component.name === ComponentNames.WHO_IS_ONLINE;
+    const presenceNotAdded = !this.activeComponents.includes(ComponentNames.PRESENCE);
+
+    if (isAddingWhoIsOnline && presenceNotAdded) {
+      this.realtime.presenceMouseParticipantJoinedObserver.subscribe(() => {
+        this.addComponent(component);
+      });
+      return false;
+    }
+
```

Comment:
This shouldn't be here because it spreads the logic of a component over multiple places in the application.

Another point is that due to some edge cases, such as the order in which the client adds the components, you do not need to have presence started to start who-is-online. You can simply add who-is-online and keep it on the local participant only until another participant joins the presence channel. Or, when you start who-is-online, you also start the presence channel.

In the future, the who-is-online will also show the participants in 3D. 

```suggestion


```